internal/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/internal/issue.md

```diff
@@ -0,0 +1,45 @@
+# OOM Issue Diagnosis - Ktor Darwin Engine
+
+## Problem Summary
+
+Out of Memory (OOM) error occurring during PowerSync sync operations when using the Kotlin Multiplatform SDK from Swift.
+
+## Root Cause
+
+The Ktor Darwin engine's `toByteArray()` function is loading entire HTTP response bodies into memory before processing them. This happens in two steps:
+
+1. **NSURLSession accumulates all response chunks** into a single `NSData` object
+2. **Ktor converts the entire NSData to ByteArray** - allocating another full copy
```

Comment:
This isn't what actually happens - if it were the case, streaming sync would be completely broken. We still see individual lines being emitted by ktor, it's just that backpressure doesn't reach the server.